### PR TITLE
add tests for recent omit-proofs issues

### PR DESCRIPTION
**Pull request by hendriktews**

### CODE REVIEW

**Comment by hendriktews** [comment]
@erikmd , @Matafou : I am pretty sure this change is fine, I'll merge if there is no veto until Friday (or there is an approval).

**Review by erikmd — APPROVED** [review]
Thanks @hendriktews, the PR looks very good to me.

**Review by Scutua — APPROVED** [review]

**Pull request merged**
